vendor/revisions.json

```diff
@@ -1,5 +1,5 @@
 {
-  "postgres-v16": "3946b2e2ea71d07af092099cb5bcae76a69b90d6",
-  "postgres-v15": "64b8c7bccc6b77e04795e2d4cf6ad82dc8d987ed",
-  "postgres-v14": "a7b4c66156bce00afa60e5592d4284ba9e40b4cf"
+  "postgres-v16": "19ca28b1a6a38b4fddce9b665dc628b38370d49c",
```

Comment:
I also prefer to minimise changes in Postgres core and move to the extension as much as possible.
Just want to notice that Postgres changes in this PR affect mostly code which is Neon specific in any case.
Certainly we can move `RestoreRunningXactsFromClog` to extension. But IMHO it will only complicates logic and maintenance. And moving all startup logic to extension seems to require even more changes in core than we have now.

And even ifs it is really possible such refactoring seems to be out of the scope of this PR.
IMHO amount of core changes added by this PR is not so critical.